tests/generate_docs/test_generate_docs.py

```diff
@@ -0,0 +1,70 @@
+"""Tests for docs generation script"""
+import pathlib
+from subprocess import CalledProcessError
+from unittest import mock, TestCase
+
+from pyfakefs.fake_filesystem_unittest import Patcher
+
+from mbed_tools_ci.generate_docs import (
+    _clear_previous_docs,
+    _generate_pdoc_command_list,
+    generate_docs,
+)
+
+
+class TestGenerateDocs(TestCase):
+    """Tests script for generating docs using Pdoc."""
+
+    def test_clear_previous_docs(self):
+        with Patcher() as patcher:
+            fake_output_dir = pathlib.Path("local_docs")
+            patcher.fs.create_file(
+                str(fake_output_dir.joinpath("some_docs_file.html")),
+                contents="This is some old documentation.",
+            )
+            self.assertTrue(fake_output_dir.is_dir())
+
+            _clear_previous_docs(fake_output_dir)
+            self.assertFalse(fake_output_dir.is_dir())
+
+    def test_clear_previous_docs_none_exist(self):
+        """Clearing old docs shouldn't error if none exist."""
+        fake_output_dir = pathlib.Path("local_docs")
+        self.assertFalse(fake_output_dir.is_dir())
+
+        _clear_previous_docs(fake_output_dir)
+
+    @mock.patch("mbed_tools_ci.generate_docs._clear_previous_docs")
+    @mock.patch("mbed_tools_ci.generate_docs.check_call")
+    def test_generate_docs(self, check_call, _clear_previous_docs):
+        """Given an output path, it should generate docs into that location.
+
+        And also return 0.
+        """
+        fake_output_dir = pathlib.Path("fake/docs")
```

Comment:
It's probably better to do something like `pathlib.Path("fake", "docs")` I'm not sure `pathlib` will resolve this form of argument correctly cross platform.

Reply:
Ah thats interesting - that means I have misunderstood where I use it in other places too.

Reply:
it does on Windows from what I can see on my machine:
```
>>> import pathlib
>>> pathlib.Path('test/path')
WindowsPath('test/path')
>>> a=pathlib.Path('test/path')
>>> str(a)
'test\\path'
```

Reply:
Oh, good to know. I wasn't sure if `pathlib` was smart enough to do that. Seems as though it is!

Reply:
👍 Thanks for checking for me @acabarbaye!